Backend/BlockChangesService.cs

```diff
@@ -0,0 +1,72 @@
+using FRITeam.Swapify.Backend.Interfaces;
+using FRITeam.Swapify.Entities;
+using FRITeam.Swapify.Entities.Enums;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace FRITeam.Swapify.Backend
+{
+
+    public class BlockChangesService : IBlockChangesService
+    {
+        private readonly IMongoDatabase _database;
+        private IMongoCollection<BlockChangeRequest> _blockChangesCollection => _database.GetCollection<BlockChangeRequest>(nameof(BlockChangeRequest));
+
+        public BlockChangesService(IMongoDatabase database)
+        {
+            _database = database;
+        }
+
+        public async Task AddAsync(BlockChangeRequest entityToAdd)
+        {
+            entityToAdd.Id = Guid.NewGuid();
+            await _blockChangesCollection.InsertOneAsync(entityToAdd);
+        }
+
+        public async Task<BlockChangeRequest> FindExchange(BlockChangeRequest blockRequest)
```

Comment:
For what do you need this public method?

Reply:
I will change it to private